src/app/lib/views/generic_detail.js

```diff
@@ -0,0 +1,238 @@
+(function(App) {
+    'use strict';
+
+    var loadImage = function (img, type, onLoad) {
+        var cache = new Image();
+        cache.src = img;
+
+        cache.onload = function () {
+            if (img.indexOf('.gif') !== -1) { // freeze gifs
+                var c = document.createElement('canvas');
+                var w  = c.width = img.width;
+                var h = c.height = img.height;
+
+                c.getContext('2d').drawImage(cache, 0, 0, w, h);
+                img = c.toDataURL();
+            }
+            onLoad(img);
+        };
+
+        cache.onerror = () => {
+            onLoad(null);
+        };
+    };
+
+    App.View.ActionBar = Backbone.Marionette.ItemView.extend({
+        template: '#action-bar-tpl',
+        className: 'actions-bar',
+        ui: {
+            bookmarkIcon: '.favourites-toggle',
+            watchedIcon: '.watched-toggle',
+        },
+        events: {
+            'click .go-back': 'closeDetails',
+            'mousedown .magnet-link': 'openMagnet',
+        },
+
+        closeDetails: function () {
+            App.vent.trigger('movie:closeDetail');
+        },
+
+    });
+
+    App.View.GenericDetail = App.View.Generic(Backbone.Marionette.LayoutView, {
+        template: '#generic-detail-tpl',
+        className: 'generic-detail',
+        ui: {
+            backdrop: '.backdrop',
+        },
+        initialize: function () {
+            //If a child was removed from above this view
+            this.bindAppEvent('viewstack:pop', function () {
+                if (_.last(App.ViewStack) === this.className) {
+                    Object.keys(this.keyboardShortCuts).map(k => (
+                        Mousetrap.bind(k, this.keyboardShortCuts[k])
+                    ));
+                }
+            });
+
+            //If a child was added above this view
+            this.bindAppEvent('viewstack:push', function () {
+                if (_.last(App.ViewStack) !== this.className) {
+                    Object.keys(this.keyboardShortCuts).map(Mousetrap.unbind);
+                }
+            });
+        },
+
+        onShow: function () {
+            console.log('Show movie detail (' + this.model.get('imdb_id') + ')');
+
+            App.MovieDetailView = this;
+
+            this.loadImages();
+            this.loadComponents();
+
+            $('.spinner').hide();
+        },
+
+        loadComponents: function () {
+            console.error('implement in subclasses');
```

Comment:
Yes, but if it is implemented, there is is no error. The error should be triggered by the caller, not the callee